ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+  
+  * In Akka the central abstraction for Location Transparency is the ActorRef: it is a self-contained serializable representation of an Actor location that enables other Actors to send messages to it. ActorRefs can also be sent as part of messages to introduce Actors to one another.
+  
+##### Automatic Scale-Out
+
+  * Orleans Grains can be configured as *stateless worker* which allows multiple activations to be active at any given time. The runtime manages the scaling factor in response to the workload.
+  
+  * In Akka cluster-aware routers offer the same functionality as the stateless worker mode of Orleans with the difference that here the provided abstraction exposes available resources to the program as the cluster grows but Akka itself does not contain the logic or ability to spin up additional nodes in response to increased load. This is left to other tools that manage server fleets.
+  
+#### Actor Interfaces
+
+  * The API provided by Orleans uses code generation in order to present Actor interactions exactly like normal method invocations—with the only difference that all methods are required to return their result asynchronously. Composition of results is performed using the async–await language features of C# and friends.
+  
+  * Akka distinguishes Actor interactions from normal method invocations by requiring the use of the ActorRef’s `!` or `tell` methods in order to send messages. These method return no result value, receiving a reply from the Actor entails providing an ActorRef where the response is sent to.
+
+In this area the differences between Akka and Orleans are very pronounced and mirror the different intent stated up-front: while Orleans aims for convenience and uses syntax that is familiar to OOP practitioners, Akka makes messaging very explicit and requires the user to define message classes. There is one instance of providing a similar model as async–await, namely the interaction between a PersistentActor and its Journal using the `persist` and `persistAsync` methods, but this API leads to a stronger coupling of the two components than would otherwise be the case. For a PersistentActor this is intended since it cannot function without the Journal, but for Grain interactions users now have to choose between blocking the grain in an `await` call (in a logical sense, but still!) or permitting other invocations to be processed concurrently “during” the `await` call—which leads to much of the same race conditions and problems that shared-state concurrency has. In Akka the interaction of multiple Actors is clearly demarcated by sending messages, and if and Actor ignores all incoming requests until a certain reply has been received then that is an explicit choice that the programmer makes.
```

Comment:
The important refinement that is missing here is that neither horizontal nor vertical is _the solution_, instead we must acknowledge that both are orthogonal concerns and contribute to the solution. For the definitions I’m using please refer to [the glossary of the Reactive Manifesto](http://www.reactivemanifesto.org/glossary#Failure): Errors are made by clients and need to be signaled back to them while Failures render the service unable to perform its function and need vertical help for recovery. We call that supervision, it might be called differently, but the crucial notion is that clients shall not receive Failures. The other notable fact is that stack traces are meaningless in distributed systems, which makes throwing exceptions even less appropriate for Error handling than it was in classical (local) OO programming: the service should reply with a normal value instead that denominates the error.

In any case, can we conclude that I fix the description of shared-state concurrency problems to clarify that low-level data races are removed while high-level message races remain? And I shall also create a new paragraph describing the difference in failure/error handling philosophy.